### PR TITLE
PAPP-34617 add update email action

### PR DESCRIPTION
**Pull request by kzuchowska-splunk**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [ ] Testing of all the changes has been performed (for bug fixes / features)
- [ ] The manual_readme_content.md has been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [ ] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [ ] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [ ] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [ ] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe):

## Security Considerations (REQUIRED)
- [ ] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers),
  please document them in the `manual_readme_content.md`.
- [ ] If this is a new connector or you are adding new actions
    - [ ] Please document in the `manual_readme_content.md` all methods (eg, OAuth) used to authenticate
      with the service that the connector is integrating with.
    - [ ] If any actions are unable to run on SOAR Cloud, please document this in the `manual_readme_content.md`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [ ] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

File: office365_connector.py [hunk 1]
```diff
@@ -3263,6 +3263,47 @@ def _paginator(self, action_result, endpoint, limit=None, params=None, query=Non
 
         return phantom.APP_SUCCESS, list_items
 
+    def _handle_update_email(self, param):
+        self.save_progress("In action handler for: {}".format(self.get_action_identifier()))
```
Comment (mposluszny-splunk) [inline office365_connector.py:3267]: why don't we use f-string instead of format?

File: office365_connector.py [hunk 2]
```diff
@@ -3263,6 +3263,47 @@ def _paginator(self, action_result, endpoint, limit=None, params=None, query=Non
...
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        email_addr = param["email_address"]
+        message_id = param["id"]
+
+        endpoint = "/users/{0}".format(email_addr)
+        endpoint += "/messages/{0}".format(message_id)
```
Comment (mposluszny-splunk) [inline office365_connector.py:3274]: same here, also why are we creating this variable with two operations?

File: office365_connector.py [hunk 3]
```diff
@@ -3263,6 +3263,47 @@ def _paginator(self, action_result, endpoint, limit=None, params=None, query=Non
...
+
+        category = param.get('category')
+        subject = param.get('subject')
+
+        if subject is None and category is None:
+            return action_result.set_status(phantom.APP_ERROR, "Please specify one of the email properties to update")
+
+        data = {}
```
Comment (mposluszny-splunk) [inline office365_connector.py:3282]: could be more informative, what data?

File: office365_connector.py [hunk 4]
```diff
@@ -3263,6 +3263,47 @@ def _paginator(self, action_result, endpoint, limit=None, params=None, query=Non
...
+        if category is not None:
+            category = [x.strip() for x in category.split(',')]
```
Comment (mposluszny-splunk) [inline office365_connector.py:3284]: so is this "category" or categories variable?

**Review by mposluszny-splunk — CHANGES_REQUESTED** [review]

**Review by bbielinski-splunk — DISMISSED** [review]

File: office365_connector.py [hunk 5]
```diff
@@ -3263,6 +3263,46 @@ def _paginator(self, action_result, endpoint, limit=None, params=None, query=Non
 
         return phantom.APP_SUCCESS, list_items
 
+    def _handle_update_email(self, param):
+        self.save_progress(f"In action handler for: {self.get_action_identifier()}")
+        action_result = self.add_action_result(ActionResult(dict(param)))
```
Comment (mposluszny-splunk) [inline office365_connector.py:3268]: isn't param a dict already?

File: office365.json [hunk 6]
```diff
@@ -7130,6 +7130,374 @@
             },
             "output": [],
             "versions": "EQ(*)"
+        },
+        {
+            "action": "update email",
+            "description": "Update an email on the server",
+            "verbose": "Currently, this action only updates the category and subject of an email. To set multiple categories, please pass a comma-separated list to the <b>category</b> parameter.<br>NOTE: If the user tries to update the categories, then the existing categories of the email will be replaced with the new categories provided as input.",
```
Comment (mposluszny-splunk) [inline office365.json:7137]: category -> categories parameter. And yes we'll need release_notes for that ;)

**Review by mposluszny-splunk — CHANGES_REQUESTED** [review]

File: office365_connector.py (hunk 5 above)
Comment (kzuchowska-splunk) [reply office365_connector.py:3268]: ```action_result = self.add_action_result(ActionResult(dict(param)))``` is a line that gets added by app wizard and it's the same in every action so I didn't change it. But yes, param i a dict

**Review by mposluszny-splunk — APPROVED** [review]

**Review by bbielinski-splunk — APPROVED** [review]

**Pull request merged**